wiki/dfg_nodes.md

```diff
@@ -0,0 +1,53 @@
+# Understanding Data Flow Graph (DFG) Node Labels in Flowr
+
+## Overview
+The Data Flow Graph (DFG) in Flowr represents the dependencies between different code components. Each node in the DFG contains specific labels that provide information about the operation being performed, the code span, and its relationship with other nodes.
+
+This document clarifies the meaning of these labels and how to interpret them.
+
+## Node Label Structure
+Each node in the DFG follows this general format:
+
+```
+[NodeType] (NodeID) LineRange ParentNodes
+```
+
+### **Breakdown of Components:**
+- **NodeType:** Indicates the type of operation. Example: `[RBinaryOp]` for a binary operation.
+- **NodeID:** A unique identifier for the node. Example: `(7)`.
+- **LineRange:** The span of lines in the source code that the node covers. Example: `2.1-10` (meaning the node spans lines 2.1 to 10).
```

Comment:
The compact presentation `2.1-10` tells you that the vertex spans _only_ the second line, ranging from character position 1 to 10.